vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,902 @@
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CB",
+      PARAM = "IMWG Clinical Benefit by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSORRES, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD == "CB" & AVALC == "Y"
+)
+```
+
+Similarly, we can define the parameters:
+
+* CR or better response (`PARAMCD` = `CR`),  
+* VGPR or better response (`PARAMCD` =`VGPR`).
+
+```{r}
+cr_y_imwg <- event(
+  description = "Define sCR or CR as response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("sCR", "CR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(cr_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "CRRSP",
+      PARAM = "IMWG Complete Response by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+
+vgpr_y_imwg <- event(
+  description = "Define sCR, CR or VGPR as response",
+  dataset_name = "ovr",
+  condition = AVALC %in% c("sCR", "CR", "VGPR"),
+  set_values_to = exprs(AVALC = "Y")
+)
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(vgpr_y_imwg, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "VGPRRSP",
+      PARAM = "IMWG VGPR Rate by Investigator (confirmation not required)",
+      PARCAT1 = "Investigator",
+      PARCAT2 = "IMWG",
+      AVAL = yn_to_numeric(AVALC)
+    ),
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, RSORRES, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD %in% c("CRRSP", "VGPRRSP") & AVALC == "Y"
+)
+```
+
+## Derive Best Overall Response Parameter {#bor}
+
+The function `admiral::derive_extreme_event()` can be used to derive the best
+overall response (without confirmation required) parameter.
+
+Please note that the order of the events specified for `events` is important.
+For example, a subject with `PR`, `PR`, `CR` qualifies for both `bor_cr` and
+`bor_pr`. As `bor_cr` is listed before `bor_pr`, `CR` is selected as best overall
+response for this subject.
+
+Some events such as `bor_cr`, `bor_pr` have been defined in {admiralonco} for
+RECIST 1.1. Missing events specific to IMWG criteria are defined below.  
+**Note**: *For `SD`, it is not required as for RECIST1.1 that the response occurs after a protocol-defined number of days.*
```

Comment:
I think we don't need to mention something here about RECIST1.1?

Reply:
updated in line 834